src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -802,6 +802,57 @@ static int _xccdf_item_recursive_gather_selected_rules(struct xccdf_policy *poli
 	return ret;
 }
 
+static char *_comment_multiline_text(char *text)
+{
+	if (text == NULL) {
+		return oscap_strdup("Not available");
+	}
+	const char *filler = "\n# ";
+	size_t buffer_size = strlen(text) + 1; // +1 for terminating '\0'
+	char *buffer = malloc(buffer_size);
+	char *saveptr, *str;
+	size_t filler_len = strlen(filler);
+	size_t result_len = 0;
+	bool first = true;
+	for (str = text; ; str = NULL) {
+		char *token = oscap_strtok_r(str, "\n", &saveptr);
+		if (token == NULL) {
+			break;
+		}
+		while (isspace(*token)) {
+			token++;
+		}
+		size_t token_len = strlen(token);
+		if (token_len > 0) {
+			char *token_last_char = token + token_len - 1;
+			while(isspace(*token_last_char)) {
+				token_last_char--;
+			}
+			*(token_last_char + 1) = '\0';
+			token_len = strlen(token);
+		}
+		if (token_len > 0) {
```

Comment:
This is also a good candidate for a function for extraction.

Reply:
It doesn't seem resuable to me because there is multiple assumptions (size of the buffer, size of the filler, size of the token, index of the token).

Reply:
It may not be reusable, but it would decrease the cognitive complexity of the code. If you are against an extraction, then put a comment there.

Reply:
I think we can survive coding normally it this case. We don't need to create lot of pointers if we are not going to use the code again.

Reply:
OK, may be.